spec/ddtrace/contrib/concurrent_ruby/integration_test_spec.rb

```diff
@@ -46,8 +46,8 @@
       deferred_execution
     end
 
-    it 'creates outer span with nil parent' do
-      expect(outer_span.parent).to be_nil
+    it 'creates outer span without a parent' do
+      expect(outer_span.parent_id).to eq(0)
     end
```

Comment:
I wonder if it would make sense to explicitly name this concept; e.g. create `Span#root_span?`.

Then, we could make all these tests a lot more explicit: `expect(outer_span).to be_root_span`

---

We could even go as far as also having `Span#has_parent?(span) = parent_id == span.span_id` and then also do `expect(some_span).to have_parent?(another_span)`.

Reply:
I added `:be_root_span`.

Regarding `have_parent?`, I'd rather not expand our public API (by adding `Span#has_parent?`). And because we can't make it private, and package visibility is not a thing in Ruby, I'm not sure we have it as suggested.

Would you think it makes sense do add a custom `has_parent?` RSpec patcher instead?

Reply:
> Would you think it makes sense do add a custom has_parent? RSpec patcher instead?

Yeah I guess it would be a nice to have.